src/entity/sbk_omo/E307C0.c

```diff
@@ -1,19 +1,110 @@
 #include "common.h"
+#include "effects.h"
 
+// size unknown
+typedef struct structE307C0 {
+    /* 0x00 */ s8 unk_00;
+    /* 0x01 */ u8 unk_01;
+    /* 0x02 */ s8 unk_02;
+    /* 0x03 */ s8 unk_03;
+    /* 0x04 */ s8 unk_04;
+    /* 0x05 */ s8 unk_05;
+    /* 0x08 */ f32 unk_08;
+    /* 0x0C */ f32 unk_0C;
+    /* 0x10 */ f32 unk_10;
+} structE307C0;
+
+extern u8 D_802BCAA0_E313F0[];
+
+s32 func_802BBE70_E307C0(void);
 INCLUDE_ASM(s32, "entity/sbk_omo/E307C0", func_802BBE70_E307C0);
 
-INCLUDE_ASM(s32, "entity/sbk_omo/E307C0", func_802BC2BC_E30C0C);
+void func_802BC2BC_E30C0C(Entity* entity) {
+    u16 currentFloor = gCollisionStatus.currentFloor;
+    structE307C0* data = (structE307C0*)entity->dataBuf;
+    PlayerStatus* playerStatus = &gPlayerStatus;
+    s32 actionState = playerStatus->actionState;
+    f32 x,y,z,sp2C;
+    s32 result = 0;
+
+    if ((currentFloor & 0x4000) && ((currentFloor & 0xFF) == entity->listIndex) && (actionState == ACTION_STATE_HAMMER)) {
+        x = playerStatus->position.x;
+        y = playerStatus->position.y + 5.0f;
+        z = playerStatus->position.z;
+        sp2C = 10.0f;
+
+        add_vec2D_polar(&x, &z, 10.0f, func_800E5348());
+        if (npc_raycast_down_sides(0x10000, &x, &y, &z, &sp2C) != 0) {
+            if (D_8010C978 & 0x4000) {
+                result = get_entity_type(D_8010C978) == 0x31;
+            }
+        }
+    } else if ((entity->collisionFlags & 1) && ((actionState == ACTION_STATE_GROUND_POUND) || (actionState == ACTION_STATE_ULTRA_POUND))) {
+        result = 1;
+    } else if (entity->collisionFlags & 0x40) {
+        result = -1;
+    }
+
+    if (result != 0) {
+        data->unk_02 &= ~1;
+        playFX_18(3, entity->position.x, entity->position.y + 35.0f, entity->position.z, 0, -1.0f, 0, 3);
+        if (result > 0) {
+            data->unk_02 |= 1;
+        }
+        entity->position.y -= 2.0f;
+        exec_entity_commandlist(entity);
+        data->unk_01 = 4;
+        disable_player_static_collisions();
+        playerStatus->unk_C5 = 5;
+    }
+}
+
+void func_802BC4B8_E30E08(Entity* entity) {
+    structE307C0* data = (structE307C0*)entity->dataBuf;
+    s32 i;
 
-INCLUDE_ASM(s32, "entity/sbk_omo/E307C0", func_802BC4B8_E30E08);
+    data->unk_04--;
+    if (data->unk_04 <= 0) {
+        i = data->unk_03;
+        if (D_802BCAA0_E313F0[i] != 0xFF) {
```

Comment:
Does this match? It might not since the array is `u8` but its worth a try
```suggestion
        if (D_802BCAA0_E313F0[i] != -1) {
```

Reply:
it doesn't